### PR TITLE
fix bug: validate auth info before building a image #108

### PR DESCRIPTION
**Pull request by ly-wjj**

### CODE REVIEW

File: image/distributionutil/registry.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ func NewV2Repository(named reference.Named, actions ...string) (distribution.Rep
 	authConfig, err := utils.GetDockerAuthInfoFromDocker(named.Domain())
 	if err != nil {
 		logger.Warn("failed to get auth info, err: %s", err)
+		return nil, err
```
Comment (fanux) [inline image/distributionutil/registry.go:32]: Sometimes we don't need login the registry when we build CloudImage if Kubefile only has COPY command. @kakaZhou719 double check this PR

File: image/distributionutil/registry.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ func NewV2Repository(named reference.Named, actions ...string) (distribution.Rep
 	authConfig, err := utils.GetDockerAuthInfoFromDocker(named.Domain())
 	if err != nil {
 		logger.Warn("failed to get auth info, err: %s", err)
```
Comment (justadogistaken) [inline image/distributionutil/registry.go:31]: This is not the point.
sometimes we do not need to login to registry(like sealer pull).

**Comment by fanux** [comment]
@justadogistaken but the build will failed if Kubefile has CMD or RUN command on push. Not a best practice. So we rellay need login on CLOUD build mod @kakaZhou719

**Comment by kakaZhou719** [comment]
@justadogistaken ，sealer pull private image will be successful ?  if so , i think we do not need to login to registry except its cloud build scenario. because cloud build will push images to registry by default, docker.json will be sent to cloud before cloud build start.@fanux.

**Comment by kakaZhou719** [comment]
@ly-wjj , as we discussed, the better way we fix this bug is , checking the docker.json file before cloud build. its necessary to do precheck for cloud build.

**Pull request closed without merging**